phoenix-core/src/it/java/org/apache/phoenix/end2end/AlterAddCascadeIndexIT.java

```diff
@@ -0,0 +1,353 @@
+        }
+    }
+
+    // Test with CASCADE INDEX <index_name>, <index_name>
+    @Test
+    public void testAlterDBOAddCascadeTwoColsTwoIndexes() throws Exception {
+        ColumnInfo [] columnArray =  {new ColumnInfo("new_column_1", PFloat.INSTANCE.getSqlType()),
+                new ColumnInfo("new_column_2", PDouble.INSTANCE.getSqlType())};
+        ColumnInfo [] columIndexArray =  {new ColumnInfo("0:new_column_1", PDecimal.INSTANCE.getSqlType()),
+                new ColumnInfo("0:new_column_2", PDecimal.INSTANCE.getSqlType())};
+
+        String query = "ALTER " + (isViewIndex ? "VIEW " : "TABLE ")
+                + phoenixObjectName + " ADD new_column_1 FLOAT, new_column_2 DOUBLE CASCADE INDEX " + indexNames;
+        conn.createStatement().execute(query);
+        if(isViewIndex) {
+            assertDBODefinition(conn, phoenixObjectName, PTableType.VIEW, 7, columnArray, false);
+            assertDBODefinition(conn, fullIndexNameOne, PTableType.INDEX, 6, columIndexArray, false);
+            assertDBODefinition(conn, fullIndexNameTwo, PTableType.INDEX, 6, columIndexArray, false);
+        } else {
+            assertDBODefinition(conn, phoenixObjectName, PTableType.TABLE, 5, columnArray, false);
+            assertDBODefinition(conn, fullIndexNameOne, PTableType.INDEX, 5, columIndexArray, false);
+            assertDBODefinition(conn, fullIndexNameTwo, PTableType.INDEX, 5, columIndexArray, false);
+        }
+
+    }
+
+    // Exception for invalid grammar
+    @Test
+    public void testAlterDBOException() throws SQLException {
+
+        String query = "ALTER " + (isViewIndex ? "VIEW " : "TABLE ") + phoenixObjectName + " ADD new_column VARCHAR ALL";
+        try {
+            conn.createStatement().execute(query);
+        } catch (Exception e) {
+            assertTrue(e.getMessage().contains(SYNTAX_ERROR));
+        }
+
+        query = "ALTER " + (isViewIndex ? "VIEW " : "TABLE ") + phoenixObjectName
+                + " ADD new_column VARCHAR CASCADE " + indexNames.split(",")[0];
+        try {
+            conn.createStatement().execute(query);
+        } catch (Exception e) {
+            assertTrue(e.getMessage().contains(SYNTAX_ERROR));
+        }
+
+        query = "ALTER " + (isViewIndex ? "VIEW " : "TABLE ") + phoenixObjectName
+                + " ADD new_column VARCHAR INDEX " + indexNames.split(",")[0];
+        try {
+            conn.createStatement().execute(query);
+        } catch (Exception e) {
+            assertTrue(e.getMessage().contains(SYNTAX_ERROR));
+        }
+
+        query = "ALTER " + (isViewIndex ? "VIEW " : "TABLE ")
+                + phoenixObjectName + " ADD new_column_1 DOUBLE CASCADE INDEX INCORRECT_NAME";
+        try {
+            conn.createStatement().execute(query);
+        } catch (Exception e) {
+            assertTrue(e.getMessage().contains(SQLExceptionCode.INCORRECT_INDEX_NAME.getMessage()));
+        }
+
+        String localIndex = generateUniqueName();
```

Comment:
Let's also have one with a mix of a global and local index to make sure that we still throw an exception.